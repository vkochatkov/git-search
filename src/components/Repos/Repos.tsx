import React from 'react';
import { ReposProps } from '../../interfaces/interface';

export const Repos: React.FC<ReposProps> = ({ repos }: ReposProps) => {
  return (
    <>
      {repos.map((repo) => (
        <div className={'card mb-3'} key={repo.id}>
          <div className="card-body">
            <h5>
              <a href={repo.html_url} target={'_blank'} rel="noreferrer">
                {repo.name}
              </a>
            </h5>
          </div>
        </div>
      ))}
    </>
  );
};
