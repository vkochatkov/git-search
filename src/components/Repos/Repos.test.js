import React from 'react';
import { Repos } from './Repos';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

configure({ adapter: new Adapter() });

const setUp = () => {
  const props = {
    repos: [
      {
        id: 2,
        html_url: 'https://avatars2.githubusercontent.com/u/39588133?v=4',
        name: '"fafaf"',
      },
    ],
  };
  const wrapper = shallow(<Repos {...props} />);

  return { props, wrapper };
};

describe('Repos component', () => {
  it('should render Repos component', () => {
    const component = setUp();
    expect(component).toMatchSnapshot();
  });
});
