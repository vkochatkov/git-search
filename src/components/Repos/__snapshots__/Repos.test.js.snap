// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Repos component should render Repos component 1`] = `
Object {
  "props": Object {
    "repos": Array [
      Object {
        "html_url": "https://avatars2.githubusercontent.com/u/39588133?v=4",
        "id": 2,
        "name": "\\"fafaf\\"",
      },
    ],
  },
  "wrapper": <Fragment>
    <div
      className="card mb-3"
      key="2"
    >
      <div
        className="card-body"
      >
        <h5>
          <a
            href="https://avatars2.githubusercontent.com/u/39588133?v=4"
            rel="noreferrer"
            target="_blank"
          >
            "fafaf"
          </a>
        </h5>
      </div>
    </div>
  </Fragment>,
}
`;
